docs/focus-visible.md

```diff
@@ -0,0 +1,70 @@
+# Accessibility focus visible
+
+One of the core pillars for supporting accessibility (a11y) within a modern web application is through maintaining the state of focus as a user tabs through an interface. The current implementation of this a11y state is via the `:focus` state introduced in CSS Level 2 (Revision 1).
+
+For a11y reasons, according to the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus#Accessibility_Concerns), it is advised to NEVER remove the browsers default settings without replacing it with a sufficient substitute.
+
+```css
+:focus { outline: none; }
+```
+
+> Never just remove the focus outline (visible focus indicator) without replacing it with a focus outline that will pass [WCAG 2.1 SC 1.4.11 Non-Text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)
+
+## Design concerns
+
+In contrast to the a11y issue, there is typically a design concern with the `:focus` state as an element within a web app has focus regardless of tab, click or tap. As a result of that, unless there is a requirement to maintain a11y the `:focus` state is typically disabled.
```

Comment:
Some awkward phrasing here. Maybe something like:

Since the `:focus` state also applies to an element when it is clicked or tapped, design will often want to remove the outline on focus. However, this is not an option if you want your project to be accessible. This presents an interesting design challenge with many opportunities.